Automattic/wp-calypso
# WPCOM Block Editor: Update editor save click tracks event

## Changes proposed in this Pull Request

* Add a tracks event when the post editor, page editor, or site editor save button is clicked

## Screenshots

### Page Editor
<img width="1368" alt="Screen Shot 2022-01-25 at 10 46 32 AM" src="https://user-images.githubusercontent.com/5414230/151039800-57633959-8c85-4b7a-bf58-9ec1fad6e218.png">

### Post Editor
<img width="1368" alt="Screen Shot 2022-01-25 at 10 46 00 AM" src="https://user-images.githubusercontent.com/5414230/151039794-aa3d11ca-002d-4f21-b11f-42307cf8768b.png">

### Site Editor
<img width="1366" alt="Screen Shot 2022-01-25 at 10 47 02 AM" src="https://user-images.githubusercontent.com/5414230/151039781-08854788-3163-4a0f-b20f-1d7b2c345124.png">

## Testing instructions



1. Sandbox a site and wp widgets
2. Run `install-plugin.sh wpcom-block-editor update/block-editor-save-click-tracks-event` in the sandbox
3. Navigate to the site editor
4. Make modifications to a template part
5. Open the dev console and switch to the `Network Requests` tab
6. Filter requests by `t.gif` 
7. Save the changes to the editor
8. Verify that a tracking `t.gif` event was sent for `wpcom_block_editor_save_click`
9. Repeat steps 3 to 8 for the page editor and the post editor



Related to #
